webui/react/src/components/FilterForm/components/FilterField.tsx

```diff
@@ -222,20 +261,40 @@ const FilterField = ({
             });
           }}
         />
-        {isSpecialColumn ? (
-          <div onKeyDownCapture={captureEnterKeyDown}>
-            <Select
-              data-test="special"
-              options={getSpecialOptions(field.columnName as SpecialColumnNames)}
-              value={fieldValue ?? undefined}
-              width={'100%'}
-              onChange={(value) => {
-                const val = value?.toString() ?? null;
-                updateFieldValue(field.id, val);
-              }}
-              onDropdownVisibleChange={setInputOpen}
-            />
-          </div>
+        {columnType !== COLUMN_TYPE.NormalColumnType ? (
+          <>
+            {columnType === COLUMN_TYPE.StringMetadataColumnType ? (
+              // StringMetadataColumnType
+              <InputSelect
+                customFilter={(options, filterValue) =>
+                  options.filter((opt) => opt.includes(filterValue))
+                }
+                data-test="special"
+                options={metadataValues.getOrElse([])}
+                value={typeof fieldValue === 'string' ? fieldValue : undefined}
+                width={'100%'}
+                onChange={(value) => {
+                  updateFieldValue(field.id, value);
+                }}
+                onDropdownVisibleChange={setInputOpen}
+              />
+            ) : (
+              // SpecialColumnType
+              <div onKeyDownCapture={captureEnterKeyDown}>
+                <Select
+                  data-test="special"
+                  options={getSpecialOptions(field.columnName as SpecialColumnNames)}
+                  value={fieldValue ?? undefined}
+                  width={'100%'}
+                  onChange={(value) => {
+                    const val = value?.toString() ?? null;
+                    updateFieldValue(field.id, val);
+                  }}
+                  onDropdownVisibleChange={setInputOpen}
+                />
+              </div>
+            )}
+          </>
```

Comment:
the fragment is unnecessary
```suggestion
            (columnType === COLUMN_TYPE.StringMetadataColumnType) ? (
              // StringMetadataColumnType
              <InputSelect
                customFilter={(options, filterValue) =>
                  options.filter((opt) => opt.includes(filterValue))
                }
                data-test="special"
                options={metadataValues.getOrElse([])}
                value={typeof fieldValue === 'string' ? fieldValue : undefined}
                width={'100%'}
                onChange={(value) => {
                  updateFieldValue(field.id, value);
                }}
                onDropdownVisibleChange={setInputOpen}
              />
            ) : (
              // SpecialColumnType
              <div onKeyDownCapture={captureEnterKeyDown}>
                <Select
                  data-test="special"
                  options={getSpecialOptions(field.columnName as SpecialColumnNames)}
                  value={fieldValue ?? undefined}
                  width={'100%'}
                  onChange={(value) => {
                    const val = value?.toString() ?? null;
                    updateFieldValue(field.id, val);
                  }}
                  onDropdownVisibleChange={setInputOpen}
                />
              </div>
            )
```

Reply:
its necessary. it shows error without the fragment.

Reply:
I'm unable to reproduce that locally. If you think about it, does it make sense? A fragment is used to contain multiple react children, but here, we're containing one child that could be one or two things.

Reply:
fixed